### DISCUSSION TITLE
GeoJSON map API / выдача меток в виде geoJSON

### DISCUSSION THREAD

**Discussion by mkgrgis** [PastVu/pastvu#364] [Ideas / Идеи и предложения]
Get a point of photographer or painter as geoJSON.

При создании архиватора точек столкнулся с тем, что API сейчас нацелено строго на программное "разбавление" данных из MongoDB на веб-странце данного изображения, что логично для БД сайта, но странновато для API. 
Пердагаю изменить формат выдачи для упрощения работы `leaflet.js`, облегчения экспорта в PostGIS, а также в перспективе появления точек и более сложных фигур изображённых объектов наряду с уже имеющимися точками съёма изображения.
Предлагаю, исполняя спецификацию geoJSON, данные результата/результатов пространственного поиска оформлять как geoJSON, где в ветви `properties` размещать все прочие, не географические данные в формате, близком к существующему.

**Comment by paul-k-pastvu** [comment]
> API сейчас нацелено строго на программное "разбавление" данных из MongoDB на веб-странце данного изображения

А можно вот это чуть пояснить? Как сейчас, как будет, и что изменится для пользователей API (RetroMap)?

**Reply by mkgrgis** [reply]
Сейчас выходит документ json с компонентом внутри `"geo": [56.165548, 101.699074],` Этот компонент может быть только точкой. Формат вывода не регулируется никаким стандартом.
Предлагается:
```
{
      "type": "Feature",
      "properties": { ДОКУМЕНТ JSON },
      "geometry": {
        "type": "Point",
        "coordinates": [
          30.450410842895508,
          59.689541779722106
        ]
      }
    }
}
```
когда сможем выводить не только точки, но и сектора или линии, то API будет полностью готов выдавать и их с возможной аналитикой пересечения видов. Будет, например, `"type": "Polygon"`.
geoJSON - гарнтированно наибыстрейший формат при роботе с `leaflet.js` и удобен для экспорта в PostGIS.
В properties можем закладывать также цвет и параметры маркера.

**Comment by mkgrgis** [comment]
Новое устройство [пространственных индексов MongoDB](https://docs.mongodb.com/manual/core/2dsphere/) явно стимулирует работу с geoJSON. См. также https://github.com/PastVu/pastvu/pull/369/commits/c9419c876bae2344017bfcec60f570000b8be44e
Итого: MongoDB и leaflet.js склоняют основную часть кода и API к работе с geoJSON вместо простой пары координат с названиями lon/lat. В основном коде нужно будет проверить только редактирование geoJSON point имеющимися средствами. Тогда указание второй точки, то есть точки объекта на изображении приведёт только к расширению возможностей редактирования geoJSON.
